From a9c1620b1768234a7695ba8cee129e83a11ab156 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Mon, 21 Oct 2024 15:22:41 +0200
Subject: [PATCH 1308/1695] plat-stm32mp2: adapt SRAM1 mapping for M33 TDCID

If Cortex-M33 is TDCID, the bsec_mirror located in the first 4kB of
SRAM1 should be mapped as non-secure.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b5bfa6ad88b0ab8f7dfdab34a36ced626ae66fa5 plat-stm32mp2: adapt SRAM1 mapping for M33 TDCID]
Change-Id: I36f157f74ce45d39fa5d1f24b16126a4a445807f
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/412910
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/plat-stm32mp2/main.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/main.c b/core/arch/arm/plat-stm32mp2/main.c
index 77c3fdfec..f4e14236c 100644
--- a/core/arch/arm/plat-stm32mp2/main.c
+++ b/core/arch/arm/plat-stm32mp2/main.c
@@ -60,8 +60,13 @@ register_phys_mem_pgdir(MEM_AREA_IO_SEC, GIC_BASE, GIC_SIZE);
 
 register_phys_mem_pgdir(MEM_AREA_IO_NSEC, DBGMCU_BASE, DBGMCU_SIZE);
 
+#ifdef CFG_STM32_CM33TDCID
+/* Map beginning SRAM1 as read-only non-secure for BSEC shadow */
+register_phys_mem_pgdir(MEM_AREA_RAM_NSEC, SRAM1_BASE, SIZE_4K);
+#else
 /* Map beginning SRAM1 as read write for BSEC shadow */
 register_phys_mem_pgdir(MEM_AREA_RAM_SEC, SRAM1_BASE, SIZE_4K);
+#endif
 
 #define _ID2STR(id)		(#id)
 #define ID2STR(id)		_ID2STR(id)
-- 
2.47.2

